webapp/channels/src/components/interactive_dialog/interactive_dialog.tsx

```diff
@@ -1,51 +1,61 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
+import type {DialogElement as TDialogElement} from '@mattermost/types/integrations';
+
 import {
     checkDialogElementForError,
     checkIfErrorsMatchElements,
 } from 'mattermost-redux/utils/integration_utils';
 
 import SpinnerButton from 'components/spinner_button';
 
+import type EmojiMap from 'utils/emoji_map';
 import {localizeMessage} from 'utils/utils';
 
-import DialogElement from './dialog_element';
+import DialogElement from './dialog_element/dialog_element';
 import DialogIntroductionText from './dialog_introduction_text';
 
-export default class InteractiveDialog extends React.PureComponent {
-    static propTypes = {
-        url: PropTypes.string.isRequired,
-        callbackId: PropTypes.string,
-        elements: PropTypes.arrayOf(PropTypes.object),
-        title: PropTypes.string.isRequired,
-        introductionText: PropTypes.string,
-        iconUrl: PropTypes.string,
-        submitLabel: PropTypes.string,
-        notifyOnCancel: PropTypes.bool,
-        state: PropTypes.string,
-        onExited: PropTypes.func,
-        actions: PropTypes.shape({
-            submitInteractiveDialog: PropTypes.func.isRequired,
-        }).isRequired,
-        emojiMap: PropTypes.object.isRequired,
+export type Props = {
+    url: string;
+    callbackId?: string;
+    elements: TDialogElement[];
+    title: string;
+    introductionText?: string;
+    iconUrl?: string;
+    submitLabel?: string;
+    notifyOnCancel?: boolean;
+    state?: string;
+    onExited: () => void;
+    actions: {
+        submitInteractiveDialog: any;
+    };
+    emojiMap: EmojiMap;
+}
+
+type State = {
+    show: boolean;
+    values: {
+        [x: string]: string;
     };
+    error: string | null;
+    errors: Record<string, unknown>;
+    submitting: boolean;
+}
 
-    constructor(props) {
+export default class InteractiveDialog extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
         super(props);
 
-        const values = {};
+        const values: Record<string, unknown> = {};
```

Comment:
Why this is unknown here? and why we need casting it whenever we set it in the `State`, looks like you are doing casting to avoid to set the correct types to things.